lib/membrane_live_audio_mixer.ex

```diff
@@ -0,0 +1,220 @@
+      ]
+    ]
+
+  @impl true
+  def handle_init(_ctx, %__MODULE__{stream_format: stream_format} = options) do
+    if options.native_mixer && !options.prevent_clipping do
+      raise("Invalid element options, for native mixer only clipping preventing one is available")
+    else
+      state =
+        options
+        |> Map.from_struct()
+        |> Map.put(:mixer_state, initialize_mixer_state(stream_format, options))
+        |> Map.put(:live_queue, LiveQueue.init(stream_format))
+
+      {[], state}
+    end
+  end
+
+  @impl true
+  def handle_playing(_context, %{stream_format: stream_format} = state) do
+    {[stream_format: {:output, stream_format}], state}
+  end
+
+  @impl true
+  def handle_pad_added(_pad, context, state) do
+    input_pads_number =
+      context.pads
+      |> Enum.filter(fn {_id, pad} -> pad.direction == :input end)
+      |> Enum.count()
+
+    if input_pads_number == 1 do
+      {[start_timer: {:initiator, state.latency}], state}
+    else
+      {[], state}
+    end
+  end
+
+  @impl true
+  def handle_stream_format(_pad, stream_format, _context, %{stream_format: nil} = state) do
+    state = %{state | stream_format: stream_format}
+    mixer_state = initialize_mixer_state(stream_format, state)
+
+    {[stream_format: {:output, stream_format}], %{state | mixer_state: mixer_state}}
+  end
+
+  @impl true
+  def handle_stream_format(_pad, stream_format, _context, %{stream_format: stream_format} = state) do
+    {[], state}
+  end
+
+  @impl true
+  def handle_stream_format(pad, stream_format, _context, state) do
+    raise(
+      RuntimeError,
+      "received invalid stream_format on pad #{inspect(pad)}, expected: #{inspect(state.stream_format)}, got: #{inspect(stream_format)}"
+    )
+  end
+
+  @impl true
+  def handle_start_of_stream(
+        Pad.ref(:input, pad_id) = pad,
+        context,
+        %{live_queue: live_queue} = state
+      ) do
+    offset = context.pads[pad].options.offset
+
+    {:ok, new_live_queue} = LiveQueue.add_queue(live_queue, pad_id, offset)
+
+    {[], %{state | live_queue: new_live_queue}}
+  end
+
+  @impl true
+  def handle_end_of_stream(Pad.ref(:input, pad_id), context, %{live_queue: live_queue} = state) do
+    {:ok, new_live_queue} = LiveQueue.remove_queue(live_queue, pad_id)
+
+    actions =
+      if all_streams_ended?(context) do
+        [{:end_of_stream, :output}]
+      else
+        []
+      end
+
+    {actions, %{state | live_queue: new_live_queue}}
+  end
+
+  @impl true
+  def handle_process(
+        Pad.ref(:input, pad_id),
+        buffer,
+        _context,
+        %{live_queue: live_queue} = state
+      ) do
+    {:ok, new_live_queue} = LiveQueue.add_buffer(live_queue, pad_id, buffer)
+
+    {[], %{state | live_queue: new_live_queue}}
+  end
+
+  @impl true
+  def handle_tick(:initiator, _context, state) do
+    {[stop_timer: :initiator, start_timer: {:timer, 20 |> Membrane.Time.milliseconds()}], state}
```

Comment:
Move 20 to a module attribute. I would avoid using `|>` for single-line pipelines